File: backend/ssbes/src/main/kotlin/ru/ssbook/ssbes/entity/Description.kt
package ru.ssbook.ssbes.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "description")
data class Description(
    @Column(name = "is_yandex")
    var isYandex: Boolean? = null,

    @Column(name = "is_to_public")
    var isToPublic: Boolean? = null,

    @Column(name = "stack_of_technology")
    var stackOfTechnology: String? = null,

    @Column(name = "what_i_do")
    var whatIDo: String? = null,

    @Column(name = "office")
    var office: String? = null,

    @Column(name = "working_time")
    var workingTime: String? = null,

    @Column(name = "seat")
    var seat: Boolean? = null,

    @Column(name = "how_often_sinks")
    var howOftenSinks: String? = null,

    @Column(name = "is_open_space")
    var isOpenSpace: Boolean? = null,

    @Column(name = "work_after")
    var workAfter: Boolean? = null,

    @Column(name = "average_age")
    var averageAge: Int? = null,

    @Column(name = "is_healthy_lifestyle")
    var isHealthyLifestyle: Boolean? = null,

    @Column(name = "party")
    var party: String? = null,

    @Column(name = "smth_else")
    var smthElse: String? = null

){
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    lateinit var id: UUID
}

File: frontend/lib/src/entities/description.dart
import 'package:json_annotation/json_annotation.dart';
part 'description.g.dart';


@JsonSerializable()
class Description{
  // @JsonKey(name: 'id')
  // final String id;

  @JsonKey(name: 'is_yandex')
  final bool? isYandex;
  @JsonKey(name: 'is_to_public')
  final bool? isToPublic;
  @JsonKey(name: 'stack_of_technology')
  final String? stackOfTechnology;
  @JsonKey(name: 'what_ido')
  final String? whatIDo;
  @JsonKey(name: 'office')
  final String? office;
  @JsonKey(name: 'working_time')
  final String? workingTime;
  @JsonKey(name: 'seat')
  final bool? haveISeat;
  @JsonKey(name: 'how_often_sink')
  final String? howOftenSinks;
  @JsonKey(name: 'is_open_space')
  final bool? isOpenSpace;
  @JsonKey(name: 'work_after')
  final bool? workAfter;
  @JsonKey(name: 'average_age')
  final int? averageAge;
  @JsonKey(name: 'is_healthy_lifestyle')
  final bool? isHealthyLifestyle;
  @JsonKey(name: 'party')
  final String? party;
  @JsonKey(name: 'smth_else')
  final String? smthElse;

  factory Description.fromJson(Map<String, dynamic> json) => _$DescriptionFromJson(json);
  Map<String, dynamic> toJson() => _$DescriptionToJson(this);

  const Description({
    //required this.id,
    this.isYandex,
    this.isToPublic,
    this.stackOfTechnology,
    this.whatIDo,
    this.office,
    this.workingTime,
    this.haveISeat,
    this.howOftenSinks,
    this.isOpenSpace,
    this.workAfter,
    this.averageAge,
    this.isHealthyLifestyle,
    this.party,
    this.smthElse
  });
}